interview-questions.md

```diff
@@ -8,36 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: This is a problem that can be easily fixed. The foreign key is simply called belongs_to so for this example in the Student model it would be belongs_to :cohort
+
+Researched answer: To fix this you would run a rails g statement for the model. then add-in your foreign key into rails g and run rails db:migrate. Then you can add has_many :students in cohort.rb and belongs_to :cohort in students.rb.
 
-Researched answer:
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The RESTful routes that must always be passed params are Show, Create, Edit, Update, and Destroy. The reason why the passed params are required is because it identifies exactly what you are trying to do. For example show(params[:id])
 
-Researched answer:
+Researched answer: Show, Create, Edit, Update, and Destroy.... I can't seem to find any resources that explains this either in the syllabus or trying to look for answers on google. At least not anything I am comprehending.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Rails new (creates a new folder with the rails structure in it), rails db:create (this creates the database),
+rails server (this starts the server allowing you to see your new app in your browser)
 
-Researched answer:
+Researched answer: Rails new, gives your folder all the files it needs to run a simple app.  rails db:create, creates a database that is empty on your local. rails server allows you to open your browser at localhost3000.
```

Comment:
The generate command is a little different that the commands you have here.  Generate will create new files within your rails application.  For instance: model, controllers, views, etc.